### PR TITLE
Update introduction.rst

### PR DESCRIPTION
**Pull request by evelynfay**

### CODE REVIEW

File: doc/sphinx/source/project/introduction.rst [hunk 1]
```diff
@@ -4,189 +4,97 @@
...
-PlaidML requires expert knowledge, and we're confident that nGraph and PlaidML 
-will make life easier for many kinds of developers: 
+We developed nGraph and integrated it with PlaidML to allow developers to accelerate deep learning performance and address the problem of scalable kernel libraries. 
+
+To address the problem of scaling backends, nGraph applies graph-level optimizations to deep learning computations and unifies computational graphs from deep learning frameworks with nGraph IR. 
+
+In conjuction with Ngraph's graph-level optimizations, PlaidML automatically applies low-level optimizations to improve deep learning performance. Additionally, PlaidML offers extensive support for various hardware targets due to its abilility to generate code in LLVM, OpenCL, OpenGL, CUDA, and Metal.
+
+Foe a backend with existing kernel libraries, nGraph can readily support the target hardware because the backend only needs to support a few primitive operations. If the hardware supports one of the coding languages supported by PlaidML, developers need to specify the machine description to support the hardare. Together, nGraph and PlaidML provide the best of both worlds.
```
Comment (indie) [inline doc/sphinx/source/project/introduction.rst]: Foe a backend?

File: doc/sphinx/source/project/introduction.rst [hunk 2]
```diff
@@ -4,189 +4,97 @@
 Introduction
 ############
 
+For making further developments in Artificial Intelligence (AI), developers need better methods to accelerate the performance of deep learning. Deep learning models are becoming more complex and the sizes of their data sets are rapidly increasing, making the deployment of scalable AI solutions a greater challenge.
```
Comment (adstraw) [inline doc/sphinx/source/project/introduction.rst]: Can we start this differently?  E.g.  "Further developments in artificial intelligence will require better methods for accelerating performance of deep learning workloads"

File: doc/sphinx/source/project/introduction.rst [hunk 3]
```diff
@@ -4,189 +4,97 @@
...
-popular solution to accelerating deep learning performance. 
+Currently, there are two standard approaches for developers to accelerate deep learning performance:
 
+**Design hardware solutions dedicated to deep learning computation**
+
+Established chip manufacturers such as Intel and hardware startups are actively developing Application Specific Integrate Circuits (ASICs) to accelerate the performance of deep learning for both training and inference.
+
+**Optimize software to accelerate performance**
+
+nGraph, an open-source graph compiler, is our solution to deliver performance via software optimization. nGraph was created to provide developers with a way to accelerate the performance of upcoming, deep learning ASICs via software and to provide a significant increase in performance for standard hardware targets such as CPUs and GPUs. For deploying scalable AI solutions, nGraph uses kernel libraries which are currently a popular method to improve deep learning performance.
```
Comment (adstraw) [inline doc/sphinx/source/project/introduction.rst]: Begin with "ngraph, an open source (no hypen) *deep learning* graph compiler"

File: doc/sphinx/source/project/introduction.rst [hunk 4]
```diff
@@ -4,189 +4,97 @@
...
-learning frameworks. These kernel libraries offer a runtime performance boost 
-on specific hardware targets through highly-optimized kernels and other 
-operator-level optimizations.
+The current state-of-the-art software solution for speeding up deep learning computation is to integrate kernel libraries such as Intel® Math Kernel Library for Deep Neural Networks (Intel® MKL DNN) and Nvidia\*'s CuDNN into deep learning frameworks. These kernel libraries offer a performance boost during runtime on specific hardware targets through highly-optimized kernels and other operator-level optimizations.
 
 However, kernel libraries have three main problems: 
 
 #. Kernel libraries do not support graph-level optimizations.
 #. Framework integration of kernel libraries does not scale.
 #. There are too many kernels to write, and they require expert knowledge.
```
Comment (adstraw) [inline doc/sphinx/source/project/introduction.rst]: require expert knowledge *to create*

File: doc/sphinx/source/project/introduction.rst [hunk 5]
```diff
@@ -4,189 +4,97 @@
...
-it's also possible to extract context semantics that make it much easier to 
-work with many of the new and emerging problems in Deep Learning including 
-larger datasets, data that must be encrypted, and data-in-motion. Our solution 
-also addresses the scalability issue with kernel libraries, the current 
-popular solution to accelerating deep learning performance. 
+Currently, there are two standard approaches for developers to accelerate deep learning performance:
 
+**Design hardware solutions dedicated to deep learning computation**
+
+Established chip manufacturers such as Intel and hardware startups are actively developing Application Specific Integrate Circuits (ASICs) to accelerate the performance of deep learning for both training and inference.
```
Comment (adstraw) [inline doc/sphinx/source/project/introduction.rst]: Remove "Established" as it exclude startups.

File: doc/sphinx/source/project/introduction.rst (hunk 3 above)
Comment (adstraw) [reply doc/sphinx/source/project/introduction.rst]: Minor nitpick:  I don't like the comma after "upcoming"

File: doc/sphinx/source/project/introduction.rst [hunk 6]
```diff
@@ -4,189 +4,97 @@
...
-graphs from various frameworks and provides a unified way to connect backends 
-for targeted hardware. From here, PlaidML or one of the nGraph transformers can 
-generate code in various forms, including LLVM, OpenCL, OpenGL, Cuda and Metal. 
-This generated code is where the low-level optimizations are automatically 
-applied.  The result is a more efficient execution that does not require any 
-manual kernel integration work for most hardware targets. 
-
-What follows here is more detail about how our solution addresses these 
-problems. 
+The nGraph Compiler stack is designed to address the first two problems. nGraph applies graph-level optimizations by taking the computational graph from a deep learning framework such as TensorFlow\* and reconstructs it with nGraph
```
Comment (adstraw) [inline doc/sphinx/source/project/introduction.rst]: reconstructs -> reconstructing

File: doc/sphinx/source/project/introduction.rst [hunk 7]
```diff
@@ -4,189 +4,97 @@
...
-
-What follows here is more detail about how our solution addresses these 
-problems. 
+The nGraph Compiler stack is designed to address the first two problems. nGraph applies graph-level optimizations by taking the computational graph from a deep learning framework such as TensorFlow\* and reconstructs it with nGraph 
+:abbr:`Intermediate Representation (IR)`. nGraph IR centralizes computational graphs from various frameworks and provides a unified way to connect backends for targeted hardware. To address the third problem, nGraph is integrated with PlaidML, *a lorem ipsum*, to generate code in LLVM, OpenCL, OpenGL, Cuda, and Metal. Low-level optimizations are automatically applied to the generated code, resulting in a more efficient execution that does not require manual kernel integration for most hardware targets. 
 
+The following three sections explore the main problems of kernel libraries in more detail and describe how nGraph addresses them.
 
-Problem: Absence of graph-level optimizations
+Problem 1: Absence of graph-level optimizations
```
Comment (adstraw) [inline doc/sphinx/source/project/introduction.rst]: It may be better to say that problem 1 is that "kernel libraries do not have graph-level optimization".  It's a little redundant but I think it is more clear.

File: doc/sphinx/source/project/introduction.rst [hunk 8]
```diff
@@ -4,189 +4,97 @@
...
 
-The computation is constructed to execute ``(A+B)*C``, but in the context of 
-nGraph, we can further optimize the graph to be represented as ``A*C``. From the 
-first graph shown on the left, the operation on the constant ``B`` can be 
-computed at the compile time (known as constant folding), and the graph can be 
-further simplified to the one on the right because the constant has value of 
-zero. Without such graph-level optimizations, a deep learning framework with a 
-kernel library will compute all operations, and the resulting execution will be 
-suboptimal. 
+The computation is constructed to execute ``(A+B)*C``, but in the context of nGraph, we can further optimize the graph to be represented as ``A*C``. From the first graph shown on the left, the operation on the constant ``B`` can be computed at the compile time (known as constant folding), and the graph can be further simplified to the one on the right because the constant has a value of zero. Without such graph-level optimizations, a deep learning framework with a kernel library will compute all operations, resulting in suboptimal execution.
```
Comment (adstraw) [inline doc/sphinx/source/project/introduction.rst]: "because the constant has a value of zero *(known as algebraic simplification)*"

File: doc/sphinx/source/project/introduction.rst [hunk 9]
```diff
@@ -4,189 +4,97 @@
...
-Integrating kernel libraries with frameworks is increasingly becoming 
-nontrivial due to the growing number of new deep learning accelerators. 
-For each new deep learning accelerator, a custom kernel library integration 
-must be implemented by a team of experts. This labor-intensive work is 
-further amplified if you want your DL accelerator to support a number of 
-different frameworks. The work must be revisited any time you upgrade or 
-expand your network's hardware. Each integration is unique to the framework 
-and its set of deep learning operators, its view on memory layout, its 
-feature set, etc.
+Due to the growing number of new deep learning accelerators, integrating kernel libraries with frameworks has become increasingly nontrivial. For each new deep learning accelerator, a custom kernel library integration must be implemented by a team of experts. This labor-intensive work is further complicated by the number of frameworks as illustrated in the diagram below using orange lines.
```
Comment (adstraw) [inline doc/sphinx/source/project/introduction.rst]: Not sure I love "nontrivial"

File: doc/sphinx/source/project/introduction.rst (hunk 1 above)
Comment (adstraw) [reply doc/sphinx/source/project/introduction.rst]: I like "given a backend"

**Comment by indie** [comment]
This `introduction.rst` file was derived from the IEEE presentation given by @adstraw and @aprocter and has been rewritten/revised many times as we strategize how to best present new and interesting content on ngraph.ai.  Any changes to the core mission / overview ought to be reviewed by them both.

**Comment by indie** [comment]
> This `introduction.rst` file was derived from the IEEE presentation given by @adstraw and @aprocter and has been rewritten/revised many times as we strategize how to best present new and interesting content on ngraph.ai. Any changes to the core mission / overview ought to be reviewed by them both.

@aprocter @tzerrell @adstraw 

Found the initial (somewhat WIP) state of the `introduction.rst` as it was, closer to when you presented : https://github.com/NervanaSystems/ngraph/blob/f5b296d03bc267bb0263cc04051df3ef536872db/doc/sphinx/source/project/introduction.rst

File: doc/sphinx/source/project/introduction.rst [hunk 10]
```diff
@@ -4,189 +4,97 @@
...
 
-PlaidML addresses the kernel explosion problem in a manner that lifts a heavy 
-burden off kernel developers. It automatically lowers networks from nGraph 
-into Tile, a :abbr:Domain-Specific Language (DSL) designed for deep learning 
-that allows developers to express how an operation should calculate tensors in
-an intuitive, mathematical form via `Stripe`_. Integration of PlaidML with 
-nGraph means extra flexibility to support newer deep learning models in the 
-absence of by-hand optimized kernels for the new operations.
-
+PlaidML uses Tile, a :abbr:Domain-Specific Language (DSL) that allows developers to express how an operation should calculate tensors in an intuitive, mathematical form. PlaidML takes user-defined Tile code along with the machine description (threads, max memory input, etc.) to automatically apply low-level optimizations. These automated optimizations do not require kernel developers to write kernel libraries, easing their burden. Integrating PlaidML with nGraph provides flexbility to support newer deep learning models in the absence of hand-optimized kernels for new operations.
```
Comment (dgkutnic) [inline doc/sphinx/source/project/introduction.rst]: PlaidML is actually moving in the direction of `Stripe` and away from `Tile`. While this is accurate pertaining to the mechanics of `Tile`, we would like to emphasize in the high-level description of PlaidML that `Stripe` is our intermediate representation (IR) that utilizes iterative optimization passes.

Please contact me directly so that we can work together on adding a better summary of `Stripe`.

File: doc/sphinx/source/project/introduction.rst [hunk 11]
```diff
@@ -4,189 +4,97 @@
...
-#. Data scientists and ML developers wishing to accelerate deep learning 
-   performance.
-#. New DL accelerator developers creating an end-to-end software stack from 
-   a deep learning framework to their silicon.  
+#. Data scientists and ML developers wishing to accelerate deep learning performance.
+#. New DL accelerator developers creating an end-to-end software stack from a deep learning framework to their silicon.  
 
 
-.. _more transistors on denser and denser circuits: https://www.intel.com/content/www/us/en/silicon-innovations/moores-law-technology.html
-.. _Stripe: https://arxiv.org/abs/1903.06498
```
Comment (dgkutnic) [inline doc/sphinx/source/project/introduction.rst]: The `Stripe` paper is pertinent to the description of PlaidML and should be retained.

File: doc/sphinx/source/project/introduction.rst (hunk 9 above)
Comment (dgkutnic) [reply doc/sphinx/source/project/introduction.rst]: Perhaps "challenging" or "constrained" is the right word here.

File: doc/sphinx/source/project/introduction.rst [hunk 12]
```diff
@@ -4,189 +4,97 @@
...
-generate code in various forms, including LLVM, OpenCL, OpenGL, Cuda and Metal. 
-This generated code is where the low-level optimizations are automatically 
-applied.  The result is a more efficient execution that does not require any 
-manual kernel integration work for most hardware targets. 
-
-What follows here is more detail about how our solution addresses these 
-problems. 
+The nGraph Compiler stack is designed to address the first two problems. nGraph applies graph-level optimizations by taking the computational graph from a deep learning framework such as TensorFlow\* and reconstructs it with nGraph 
+:abbr:`Intermediate Representation (IR)`. nGraph IR centralizes computational graphs from various frameworks and provides a unified way to connect backends for targeted hardware. To address the third problem, nGraph is integrated with PlaidML, *a lorem ipsum*, to generate code in LLVM, OpenCL, OpenGL, Cuda, and Metal. Low-level optimizations are automatically applied to the generated code, resulting in a more efficient execution that does not require manual kernel integration for most hardware targets.
```
Comment (dgkutnic) [inline doc/sphinx/source/project/introduction.rst]: PlaidML is a tensor compiler.

Please contact me directly as we have some changes we'd like to make here that need an offline discussion.

**Comment by indie** [comment]
Closing this PR since the primary changes ought to be submitted in a separate PR that does not remove attributions.

**Comment by adstraw** [comment]
Reopening as a means to continue gathering feedback.  Not ignoring @indie's feedback RE attribution.

**Comment by indie** [comment]
For writers who want to contribute docs, we have a helpful section of docs that is good to read before submitting changes: https://ngraph.nervanasys.com/docs/latest/project/doc-contributor-README.html

My assertion remains that these changes will cause fewer problems if they are addressed in two separate PRs: one that refines the Intro to nGraph and one that discusses the PlaidML changes that @dgkutnic is requesting. Each of these sections in doc has been assigned to the appropriate code owner, and each owner will be able to work with you to get their changes addressed.

I'll leave it to @aprocter to decide if or when to close this PR; I prefer to not edit a file that introduces regressions.

File: doc/sphinx/source/project/introduction.rst [hunk 13]
```diff
@@ -4,189 +4,97 @@
...
-further amplified if you want your DL accelerator to support a number of 
-different frameworks. The work must be revisited any time you upgrade or 
-expand your network's hardware. Each integration is unique to the framework 
-and its set of deep learning operators, its view on memory layout, its 
-feature set, etc.
+Due to the growing number of new deep learning accelerators, integrating kernel libraries with frameworks has become increasingly nontrivial. For each new deep learning accelerator, a custom kernel library integration must be implemented by a team of experts. This labor-intensive work is further complicated by the number of frameworks as illustrated in the diagram below using orange lines. 
 
+.. _figure-B:
+
+.. figure:: ../graphics/lorem-ipsum.png
```
Comment (aprocter) [inline doc/sphinx/source/project/introduction.rst]: Is this file present in the repo? (If so, maybe we should rename it from `lorem-ipsum.png` :))

File: doc/sphinx/source/project/introduction.rst [hunk 14]
```diff
@@ -4,189 +4,97 @@
...
-graph and reconstructs it in the nGraph IR with a few primitive nGraph 
-operations. With the unified computational graph, kernel libraries no longer 
-need to be separately integrated to each deep learning framework. Instead, the 
-libraries only need to support nGraph primitive operations, and this approach 
-streamlines integration process for the backend.  
+Each framework must be manually integrated with each hardware-specific kernel library. Additionally, each integration is unique to the framework and its set of deep learning operators, view on memory layout, feature set, etc. Each connection that needs to be made increases the amount of work, resulting in a fragile setup that is costly to maintain.
 
+nGraph solves this problem with nGraph bridges. A bridge takes a computational graph from a supported framework and reconstructs it in the nGraph IR with a few primitive nGraph operations. With a unified computational graph, kernel libraries no longer need to be separately integrated into each deep learning framework. Instead, the ibraries only need to support nGraph primitive operations, and this approach streamlines the integration process for the backend.
```
Comment (aprocter) [inline doc/sphinx/source/project/introduction.rst]: `computational graph from a supported framework`: not all frameworks use graphs as their native formalisms... it's a bit wordy, but maybe `a computational graph or similar structure`

Comment (aprocter) [reply doc/sphinx/source/project/introduction.rst]: Typo: `ibraries`

File: doc/sphinx/source/project/introduction.rst [hunk 15]
```diff
@@ -4,189 +4,97 @@
...
-need to be separately integrated to each deep learning framework. Instead, the 
-libraries only need to support nGraph primitive operations, and this approach 
-streamlines integration process for the backend.  
+Each framework must be manually integrated with each hardware-specific kernel library. Additionally, each integration is unique to the framework and its set of deep learning operators, view on memory layout, feature set, etc. Each connection that needs to be made increases the amount of work, resulting in a fragile setup that is costly to maintain.
 
+nGraph solves this problem with nGraph bridges. A bridge takes a computational graph from a supported framework and reconstructs it in the nGraph IR with a few primitive nGraph operations. With a unified computational graph, kernel libraries no longer need to be separately integrated into each deep learning framework. Instead, the ibraries only need to support nGraph primitive operations, and this approach streamlines the integration process for the backend.  
 
-Problem: Increasing number of kernels 
+
+Problem 3: Increasing the number of kernels
```
Comment (aprocter) [inline doc/sphinx/source/project/introduction.rst]: I'd drop the definite article here (the original was saying `it is a problem that the number of kernels keeps increasing`, not `it is difficult to increase the number of kernels`). Or maybe rewrite as `The number of required kernels keeps growing`

File: doc/sphinx/source/project/introduction.rst [hunk 16]
```diff
@@ -4,189 +4,97 @@
...
 -------------------------------------
 
-Kernel libraries need to be integrated with multiple deep learning frameworks, 
-and this arduous task becomes even harder due to increased numbers of required 
-kernels for achieving optimal performance. The number of required kernels is 
-product of number of chip designs, data types, operations, and the cardinality 
-of each parameter for each operation. In the past, the number of required 
-kernels was limited, but as the AI research and industry rapidly develops, the 
-final product of required kernels is increasing exponentially. 
+As previously mentioned, Kernel libraries need to be integrated with multiple deep learning frameworks, and this already arduous task becomes even harder due to the greater number of a kernels for achieving optimal performance. The number of required kernels is based on the number of chip designs, data types, operations, and the cardinality of each parameter per operation. In the past, the number of required kernels was limited, but as AI research and industry continue to rapidly develop, the number of required kernels is exponentially increasing.
```
Comment (aprocter) [inline doc/sphinx/source/project/introduction.rst]: `kernel` should not be capitalized

Comment (aprocter) [reply doc/sphinx/source/project/introduction.rst]: typo: `a kernels`

Comment (aprocter) [reply doc/sphinx/source/project/introduction.rst]: I wonder if we could rework this a bit, maybe start with something like "Past deep learning research has been built on a small set of standard computational primitives (convolution, GEMM, etc.). But as research progresses and industrial applications of deep learning continue to develop, the number of required kernels is exponentially increasing." Then go into the stuff about number of chip designs, data types, cardinality, &c.

File: doc/sphinx/source/project/introduction.rst [hunk 17]
```diff
@@ -4,189 +4,97 @@
...
-
-This documentation provides technical details of nGraph's core functionality, 
-framework and backend integrations. Creating a compiler stack like nGraph and 
-PlaidML requires expert knowledge, and we're confident that nGraph and PlaidML 
-will make life easier for many kinds of developers: 
+We developed nGraph and integrated it with PlaidML to allow developers to accelerate deep learning performance and address the problem of scalable kernel libraries. 
+
+To address the problem of scaling backends, nGraph applies graph-level optimizations to deep learning computations and unifies computational graphs from deep learning frameworks with nGraph IR. 
+
+In conjuction with Ngraph's graph-level optimizations, PlaidML automatically applies low-level optimizations to improve deep learning performance. Additionally, PlaidML offers extensive support for various hardware targets due to its abilility to generate code in LLVM, OpenCL, OpenGL, CUDA, and Metal.
```
Comment (aprocter) [inline doc/sphinx/source/project/introduction.rst]: typo: `conjuction`. `Ngraph` should be `nGraph`.

File: doc/sphinx/source/project/introduction.rst (hunk 1 above)
Comment (aprocter) [reply doc/sphinx/source/project/introduction.rst]: typo: `hardare`

File: doc/sphinx/source/project/introduction.rst (hunk 3 above)
Comment (dgkutnic) [reply doc/sphinx/source/project/introduction.rst]: @aprocter the reference to PlaidML is appropriate and welcome here.

**Comment by indie** [comment]
Thanks, @evelynfay for your efforts to update this.  I do appreciate them, even if my previous replies were short on pleasantries.  For the short-term, I'm focused on getting the `0.21-doc` published, and won't be able to review any additional edits you make.  

A few tips for contributors who are new to the highly-dynamic environment of docs in AI software:  

* A good place to start is "document something you figured out how to get working".  If you find and can follow the [doc-contributor-README](https://ngraph.nervanasys.com/docs/latest/project/doc-contributor-README.html) to build docs, that file has several hints at the kind of content our audience understands and is looking for content on. It's a not a narrow audience, but if you don't understand how wide the audience is, you'll inadvertently break or block things.  Content changes should be targeted to something more specific than "developers"; there are experts who work on all parts of the stack. 

* Start with something small.  It's okay to add a "patch" to fix a typo or suggest a word change; larger changes to files or structure require research and testing first, as well as some logic for why you think something needs changed.  

* Wrap at ~80 column width (min) (~100 max is okay) for all content in the nGraph core `doc` repo.  We do our best to help authors maintain and source-link their code and contributions; switching things up just to overwrite something already documented doesn't always make it better.  Please don't remove links in content.

**Comment by diyessi** [comment]
@evelynfay Can you resolve the merge conflicts? Thanks.

**Comment by diyessi** [comment]
No longer relevant.
